### ISSUE TITLE
No stacktrace when a rule throws an exception

### ISSUE THREAD

**Issue by MasseGuillaume** [scalacenter/scalafix#745]
input:

```scala
case class MyRule(index: SemanticdbIndex)  extends SemanticRule(index, "MyRule") {
  throw new Exception("boom")
  override def fix(ctx: RuleCtx): Patch = Patch.empty
```

output:

```scala
[info] - scala/path/to/test *** FAILED ***
[info]   java.lang.IllegalStateException: Type mismatch;
[info]   found    : String (value: "scala:MyRule")
[info]   expected : scalafix.rule.Rule
[info]   at metaconfig.Configured.get(Configured.scala:11)
[info]   at scalafix.testkit.DiffTest$$anonfun$$nestedInanonfun$fromSemanticdbIndex$2$1.applyOrElse(DiffTest.scala:41)
[info]   at scalafix.testkit.DiffTest$$anonfun$$nestedInanonfun$fromSemanticdbIndex$2$1.applyOrElse(DiffTest.scala:33)
[info]   at scala.collection.TraversableOnce.collectFirst(TraversableOnce.scala:145)
[info]   at scala.collection.TraversableOnce.collectFirst$(TraversableOnce.scala:132)
[info]   at scala.meta.tokens.Tokens.collectFirst(Tokens.scala:28)
[info]   at scalafix.testkit.DiffTest$.$anonfun$fromSemanticdbIndex$2(DiffTest.scala:33)
[info]   at scalafix.testkit.SemanticRuleSuite.$anonfun$runOn$1(SemanticRuleSuite.scala:94)
[info]   at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:12)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   ...
```

expected:
```scala
java.lang.Exception: boom
  // full  stack
```

**Comment by olafurpg** [comment]
What version are you using? In latest master, the error message is

```scala
[info] - test/FqnRule.scala *** FAILED *** (580 milliseconds)
[info]   java.lang.Exception: Message
[info]   at banana.rule.SemanticRuleV1.<init>(FqnRule.scala:42)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[info]   at java.lang.Class.newInstance(Class.java:442)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.toRule(RuleDecoderOps.scala:42)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.tryClassload(RuleDecoderOps.scala:48)
[info]   at scalafix.v1.RuleDecoder$.fromString(RuleDecoder.scala:58)
[info]   at scalafix.v1.RuleDecoder$$anon$1.$anonfun$read$1(RuleDecoder.scala:101)
```

If you are writing scalafix.v0 rules the error message is slightly larger but still contains the stack trace
```
[info] - test/FqnRule.scala *** FAILED *** (424 milliseconds)
[info]   java.lang.reflect.InvocationTargetException:
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.$anonfun$toRule$1(RuleDecoderOps.scala:32)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.toRule(RuleDecoderOps.scala:34)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.tryClassload(RuleDecoderOps.scala:48)
[info]   at scalafix.v1.RuleDecoder$.fromString(RuleDecoder.scala:58)
[info]   at scalafix.v1.RuleDecoder$$anon$1.$anonfun$read$1(RuleDecoder.scala:101)
[info]   at scala.collection.immutable.List.flatMap(List.scala:335)
[info]   ...
[info]   Cause: java.lang.Exception: boom
[info]   at banana.rule.FqnRule.<init>(FqnRule.scala:12)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.$anonfun$toRule$1(RuleDecoderOps.scala:32)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.toRule(RuleDecoderOps.scala:34)
[info]   at scalafix.internal.reflect.RuleDecoderOps$.tryClassload(RuleDecoderOps.scala:48)
[info]   at scalafix.v1.RuleDecoder$.fromString(RuleDecoder.scala:58)
[info]   at scalafix.v1.RuleDecoder$$anon$1.$anonfun$read$1(RuleDecoder.scala:101)
```

**Comment by MasseGuillaume** [comment]
0.5.10

**Comment by olafurpg** [comment]
Seems fixes in v0.6 then :)

**Comment by MasseGuillaume** [comment]
@olafurpg I suggest leaving the issue open until 0.6 is released.
